src/TestInfrastructure/Framework/ConfigurableVsProjectSystemHelper.cs

```diff
@@ -121,6 +121,17 @@ public void ClearProjectProperty(Project dteProject, string propertyName)
             projMock.ClearBuildProperty(propertyName);
         }
 
+        public IEnumerable<Guid> GetAggregateProjectKinds(IVsHierarchy hierarchy)
+        {
+            ProjectMock dteProject = hierarchy as ProjectMock;
+            if(dteProject == null)
+            {
+                Assert.Inconclusive($"Only expecting {nameof(ProjectMock)} type");
```

Comment:
Why Assert Inconclusive and not Assert Fail? In tests, inconclusive in used to show that you cannot run the test or you cannot tell if it passes or fails. In practice this is because the test fails to provision its data or the some integration component is not available (for example if you are testing against a live server and it is down). 

In this case, I'd argue that the mock is not used correctly so the test should just fail.